dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogInterceptor.kt

```diff
@@ -51,14 +57,127 @@ import okhttp3.OkHttpClient
  * the possibility to modify the created [io.opentracing.Span].
  *
  */
-
-class DatadogInterceptor @JvmOverloads constructor(
+open class DatadogInterceptor
+internal constructor(
     tracedHosts: List<String>,
-    tracedRequestListener: TracedRequestListener = NoOpTracedRequestListener()
-) :
-    Interceptor by CombinedInterceptor(
-        listOf(
-            TracingRequestInterceptor(tracedRequestListener, tracedHosts),
-            RumRequestInterceptor()
+    tracedRequestListener: TracedRequestListener = NoOpTracedRequestListener(),
+    localTracerFactory: () -> Tracer
+) : TracingInterceptor(tracedHosts, tracedRequestListener, localTracerFactory) {
+
+    /**
+     * Creates a [TracingInterceptor] to automatically create a trace around OkHttp [Request]s, and
+     * track RUM Resources.
+     *
+     * @param tracedHosts a list of all the hosts that you want to be automatically tracked
+     * by our APM [TracingInterceptor]. If no host provided the interceptor won't trace
+     * any OkHttpRequest, nor propagate tracing information to the backend.
+     * Please note that the host constraint will only be applied on the TracingInterceptor and we will
+     * continue to dispatch RUM Resource events for each request without applying any host filtering.
+     * @param tracedRequestListener which listens on the intercepted [okhttp3.Request] and offers
+     * the possibility to modify the created [io.opentracing.Span].
+     */
+    @JvmOverloads
+    constructor(
+        tracedHosts: List<String>,
+        tracedRequestListener: TracedRequestListener = NoOpTracedRequestListener()
+    ) : this(tracedHosts, tracedRequestListener, { AndroidTracer.Builder().build() })
+
+    // region Interceptor
+
+    /** @inheritdoc */
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val url = request.url().toString()
+        val method = request.method()
+        val requestId = identifyRequest(request)
+        println("stopResource $requestId $method $url")
+
+        GlobalRum.get().startResource(requestId, method, url)
+
+        return super.intercept(chain)
+    }
+
+    // endregion
+
+    // region TracingInterceptor
+
+    /** @inheritdoc */
+    override fun onRequestIntercepted(
+        request: Request,
+        span: Span,
+        response: Response?,
+        throwable: Throwable?
+    ) {
+        super.onRequestIntercepted(request, span, response, throwable)
+
+        if (throwable != null) {
+            handleThrowable(request, throwable)
+        } else {
+            handleResponse(request, response, span)
+        }
```

Comment:
is it impossible to have both `response` and `throwable` null?

Reply:
Well in theory nothing's impossible, but in practice no this should never happen.